packages/adapter-cloudflare/src/worker.js

```diff
@@ -53,10 +53,21 @@ const worker = {
 			});
 		}
 
-		// Writes to Cache only if allowed & specified
+		// write to Cache only if allowed & specified
 		pragma = res.headers.get('cache-control');
-		return pragma && res.ok ? Cache.save(req, res, context) : res;
+
+		// only save good responses to Cache
+		return should_cache(res) ? Cache.save(req, res, context) : res;
```

Comment:
This should just be a statusCode check, leaving `Cache.save` to employ the rest of the "is this cacheable" logic

Reply:
Thanks! You're right, `Cache.save` already looks into the `cache-control` (and also the `vary`) header, which makes `should_cache` redundant. I'll revert to the simple `is_error` check as seen in adapter-node.

Reply:
This means I can also remove the `pragma && res.ok` condition, correct? Because the response will not be `ok` if `status > 399`, and Worktop makes sure it respects the headers.

Reply:
Yes

Reply:
@lukeed I have looked into `Cache.save` and found out that it didn't handle the case when `cache-control` is empty and let it cache by default. So every dynamic content without cc will cache by default in case of performance it's good but I'm curious about security as I describe in this issue https://github.com/lukeed/worktop/issues/176.

Reply:
Can confirm. The check for a `cache-control` header got lost in this PR.

I'll create an issue and propose a fix.